apps/keyboard/js/views/swipeable_panel_view.js

```diff
@@ -105,6 +101,34 @@ SwipeablePanelView.prototype.gotoSection = function(index) {
   this._updateIndicator();
 };
 
+SwipeablePanelView.prototype.resize = function resize(totalWidth) {
+  var self = this;
+
+  this.options.totalWidth = totalWidth;
+
+  // Stop SwipingDetector while rotating
+  if (this.swipingDetector && typeof this.swipingDetector.stop === 'function') {
+    this.swipingDetector.stop();
+  }
+  // Reposition sections when resize;
+  this.sections.forEach(function(section, index) {
+    var translateX = (index === self.currentSectionIndex) ? 0 :
+    (self.currentSectionIndex > index) ? -totalWidth : totalWidth;
+
+    section.style.transform = 'translateX(' + translateX + 'px)';
+  });
+
+  this._bindSwipingDetector();
+};
+
+SwipeablePanelView.prototype._bindSwipingDetector = function() {
```

Comment:
Actually, you are stopping and starting `SwipingDetector` in the same function loop, so I don't really know what this achieves...